ddl/lightning/res_mgr.go

```diff
@@ -204,6 +202,25 @@ func (m *LightningMemoryRoot) DeleteBackendContext(bcKey string) {
 	return
 }
 
+// In exception case, clear intermediate files that lightning engine generated for index.
+func(m *LightningMemoryRoot) ClearEngines(jobId int64, indexIds ...int64) {
+    for _, indexId := range indexIds {
+        eiKey := GenEngineInfoKey(jobId, indexId)
+	    ei, exist := m.EngineMgr.engineCache[eiKey]
+	    if exist {
+		    indexEngine := ei.openedEngine
+		    closedEngine, err := indexEngine.Close(ei.backCtx.Ctx, ei.cfg)
+            if err != nil {
+			    log.L().Error(LERR_CLOSE_ENGINE_ERR, zap.String("Engine key", eiKey))
+			    return
+		    }
+		    // Here the local intermediate file will be removed.
+		    closedEngine.Cleanup(ei.backCtx.Ctx)
+	    }
+	}
+	return
+}
+
 // Check and allocate one EngineInfo
 func (m *LightningMemoryRoot) RegistEngineInfo(job *model.Job, bcKey string, engineKey string, indexId int32, workerCount int) (int, error) {
 	var (
```

Comment:
this variable *ei is only used for cal the size. did we need  define this variable?